src/Traits/TemporaryMedia.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Oxygencms\Core\Traits;
+
+use Oxygencms\Core\Models\Model;
+use Oxygencms\Core\Models\Temporary;
+
+trait TemporaryMedia
+{
+    /**
+     * Move the media from the temporary models to the created model and delete the temporary model
+     *
+     * @param Model $entity
+     * @param int $temporary_id
+     */
+    public function moveMedia(Model $entity, int $temporary_id)
+    {
+        $temporary = Temporary::find($temporary_id);
+
+        $media = $temporary->getMedia();
+
+        $media->each(function ($item) use($entity) {
+            $item->move($entity);
+        });
```

Comment:
after moving all media check if the new `$entity` has all the media assigned and all the media collections and conversions are still there and acessible